### PR TITLE
Klokane/enum samples on master

### PR DESCRIPTION
**Pull request by klokane**
rebase of #521 on master

### CODE REVIEW

**Comment by pksunkara** [comment]
Do you want to close the #521?

File: src/RefractDataStructure.cc [hunk 1]
```diff
@@ -853,6 +874,49 @@ namespace
...
+        }
+
+        auto info = Merge<T>()(std::move(values));
+
+        if (info.value.empty()) {
+            element.attributes().set(key, make_empty<EnumElement>());
+            return;
+        }
+
+        element.attributes().set(key, make_element<EnumElement>(dsd::Enum{ std::move(info.value.back()) }));
```
Comment (tjanc) [inline src/RefractDataStructure.cc]: Should we pop the last element after the move? Its value is `nullptr`.

Comment (klokane) [reply src/RefractDataStructure.cc]: We can, but in this case rest is dropped, so we can omit pop in this case
It is same like nonspecialized case. see:
https://github.com/apiaryio/drafter/blob/ce0edbd63e9ef3e8d9249929d7031666cba74b24/src/RefractDataStructure.cc#L874

**Comment by klokane** [comment]
Changelog updated

**Pull request merged**